gnss_lib_py/parsers/precise_ephemerides.py

```diff
@@ -0,0 +1,811 @@
+    sidx : int
+        Nearest index within sp3 time series around which interpolated
+        function needs to be centered
+    ipos : int
+        No. of data points from sp3 data on either side of sidx
+        that will be used for computing interpolated function
+    method : string
+        Type of interpolation method used for sp3 data (the default is
+        CubicSpline, which depicts third-order polynomial)
+
+    Returns
+    -------
+    func_satpos : np.ndarray
+        Instance with 3-D array of scipy.interpolate.interpolate.interp1d
+        that is loaded with .sp3 data
+    """
+
+    func_satpos = np.empty((3,), dtype=object)
+    func_satpos[:] = np.nan
+
+    if method=='CubicSpline':
+        low_i = (sidx - ipos) if (sidx - ipos) >= 0 else 0
+        high_i = (sidx + ipos) if (sidx + ipos) <= len(sp3data.tym) else -1
+
+        if verbose:
+            print('Nearest sp3: ', sidx, sp3data.tym[sidx], \
+                                   sp3data.xpos[sidx], sp3data.ypos[sidx], sp3data.zpos[sidx])
+
+        func_satpos[0] = interpolate.CubicSpline(sp3data.tym[low_i:high_i], \
+                                                 sp3data.xpos[low_i:high_i])
+        func_satpos[1] = interpolate.CubicSpline(sp3data.tym[low_i:high_i], \
+                                                 sp3data.ypos[low_i:high_i])
+        func_satpos[2] = interpolate.CubicSpline(sp3data.tym[low_i:high_i], \
+                                                 sp3data.zpos[low_i:high_i])
+
+    return func_satpos
+
+def extract_clk_func(clkdata, sidx, ipos = 10, \
+                     method='CubicSpline', verbose = False):
+    """Computing interpolated function over clk data for any GNSS
+
+    Parameters
+    ----------
+    clkdata : list
+        Instance of GPS-only Sp3 class list with len == # sats
+    sidx : int
+        Nearest index within sp3 time series around which interpolated
+        function needs to be centered
+    ipos : int
+        No. of data points from sp3 data on either side of sidx
+        that will be used for computing interpolated function
+    method : string
+        Type of interpolation method used for sp3 data (the default is
+        CubicSpline, which depicts third-order polynomial)
+
+    Returns
+    -------
+    func_satbias : np.ndarray
+        Instance with 1-D array of scipy.interpolate.interpolate.interp1d
+        that is loaded with .clk data
+    """
+
+    if method=='CubicSpline':
+        low_i = (sidx - ipos) if (sidx - ipos) >= 0 else 0
+        high_i = (sidx + ipos) if (sidx + ipos) <= len(clkdata.tym) else -1
+
+        if verbose:
+            print('Nearest clk: ', sidx, clkdata.tym[sidx], clkdata.clk_bias[sidx])
+
+        func_satbias = interpolate.CubicSpline(clkdata.tym[low_i:high_i], \
+                                               clkdata.clk_bias[low_i:high_i])
+
+    return func_satbias
+
+def compute_sp3_snapshot(func_satpos, cxtime, hstep = 5e-1, method='CubicSpline'):
```

Comment:
Can you rename these functions? Feels a little repetitive to have `compute_*` for a bunch of them :P